### PR TITLE
Api integration

### PR DESCRIPTION
**Pull request by srishtigautam21**
Added the API from backend for cart and wishlist management
Files to check:
LoginPage.jsx
AuthContext
CartContext
WishListContext
HorizontalCard.jsx
WishListCard.jsx
Toastify.jsx
**Please ignore that extra netlify deploy preview will delete it later on**

### CODE REVIEW

File: src/component/Context/AuthContext/AuthContext.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    e.preventDefault();
+    // console.log("login");
+
+    try {
+      const response = await axios.post("/api/auth/login", { email, password });
+      // loginUser
+      localStorage.setItem("nurishToken", response.data.encodedToken);
+      // console.log("auth", response.data.encodedToken);
+      setUserLoggedIn(true);
+      // setLoginUser({ email: "", password: "" });    //Kept this for future use
```
Comment (iRohitGaur) [inline src/component/Context/AuthContext/AuthContext.js]: suggestion: remove commented code

**Review by iRohitGaur — APPROVED** [review]
Nice work!

Code is easy to understand.

Good to merge 👍🏻

File: src/component/Context/AuthContext/AuthContext.js (hunk 1 above)
Comment (srishtigautam21) [reply src/component/Context/AuthContext/AuthContext.js]: thanks rohit i had forgot about them

File: src/component/Context/AuthContext/AuthContext.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    email: "",
+    password: "",
+  };
+  const [loginUser, setLoginUser] = useState({
+    email: "adarshbalika@gmail.com",
+    password: "adarshbalika",
+  });
+  const [signupUser, setSignUpUser] = useState(authInitialState);
+  const [isUserLoggedIn, setUserLoggedIn] = useState(false);
+  // const { email, password } = loginUser;
```
Comment (abhishektanwar) [inline src/component/Context/AuthContext/AuthContext.js]: remove comment

File: src/component/Context/AuthContext/AuthContext.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+      localStorage.setItem("nurishToken", response.data.encodedToken);
+      // console.log("auth", response.data.encodedToken);
+      setUserLoggedIn(true);
+      // setLoginUser({ email: "", password: "" });    //Kept this for future use
+      loginToast("Login Successful");
+      setTimeout(() => {
+        navigate("/products");
+      }, 200);
+    } catch (e) {
+      console.error(e);
```
Comment (abhishektanwar) [inline src/component/Context/AuthContext/AuthContext.js:38]: suggestion : We can show a toast here.

File: src/component/Context/CartContext/CartContext.js [hunk 4]
```diff
@@ -1,61 +1,34 @@
-import { useContext, createContext, useReducer } from "react";
-import { useWishList } from "../../index";
+import { useContext, createContext, useReducer, useEffect } from "react";
```
Comment (abhishektanwar) [inline src/component/Context/CartContext/CartContext.js:1]: Suggestion : We can extract out the code for reducer in a separate file and functions for api calls (increase quantity,addToCart ,deleteFromCart etc) could be extracted out to a custom hook . Will make the mode more readable .

File: src/component/Context/WishListContext/WishListContext.js [hunk 5]
```diff
@@ -1,49 +1,88 @@
...
+      );
+
+      wishListDispatch({
+        type: "ADD_TO_WISHLIST",
+        payload: response.data.wishlist,
+      });
+
+      addToWishListToast("Added to WishList");
+    } catch (e) {
+      console.error(e);
```
Comment (abhishektanwar) [inline src/component/Context/WishListContext/WishListContext.js:54]: Suggestion : we can show toast here aswell.

File: src/component/Context/WishListContext/WishListContext.js [hunk 6]
```diff
@@ -1,49 +1,88 @@
...
+        config
+      );
+
+      wishListDispatch({
+        type: "REMOVE_FROM_WISHLIST",
+        payload: response.data.wishlist,
+      });
+      deleteFromWishListToast("Removed from wishList");
+    } catch (e) {
+      console.error(e);
```
Comment (abhishektanwar) [inline src/component/Context/WishListContext/WishListContext.js:77]: suggestion : we can show toast here

File: src/component/Context/WishListContext/WishListContext.js [hunk 7]
```diff
@@ -1,49 +1,88 @@
-import { useContext, useState, createContext } from "react";
+import { useContext, useState, createContext, useReducer } from "react";
```
Comment (abhishektanwar) [inline src/component/Context/WishListContext/WishListContext.js:1]: Suggestion : Code for reducer and api calls can be extracted out .

**Review by abhishektanwar — APPROVED** [review]
Code is understandable and clean . UI also looks great .
I have made a few suggestions . Good to merge

File: src/component/Context/CartContext/CartContext.js (hunk 4 above)
Comment (srishtigautam21) [reply src/component/Context/CartContext/CartContext.js:1]: Thanks for the suggestion  abhishek will do the changes in the next PR

File: src/component/Context/AuthContext/AuthContext.js (hunk 2 above)
Comment (srishtigautam21) [reply src/component/Context/AuthContext/AuthContext.js]: done

File: src/component/Context/AuthContext/AuthContext.js (hunk 3 above)
Comment (srishtigautam21) [reply src/component/Context/AuthContext/AuthContext.js:38]: Added

File: src/component/Context/WishListContext/WishListContext.js (hunk 7 above)
Comment (srishtigautam21) [reply src/component/Context/WishListContext/WishListContext.js:1]: Thanks for suggestion will do it in next PR

File: src/component/Context/WishListContext/WishListContext.js (hunk 5 above)
Comment (srishtigautam21) [reply src/component/Context/WishListContext/WishListContext.js:54]: done

File: src/component/Context/WishListContext/WishListContext.js (hunk 6 above)
Comment (srishtigautam21) [reply src/component/Context/WishListContext/WishListContext.js:77]: Done

**Pull request merged**